api/types/registry/authconfig.go

```diff
@@ -71,6 +71,8 @@ func EncodeAuthConfig(authConfig AuthConfig) (string, error) {
 // to the caller to decide if authentication is required, and if the error can
 // be ignored.
 //
+// Deprecated: this function is deprecated and will be removed in the next release.
+//
```

Comment:
We should probably squash the first 2 commits (or first 3 with the commit I pushed), and make this a separate patch for the 28.x branch.